src/issuance/mdoc.rs

```diff
@@ -185,13 +190,17 @@ impl PreparedMdoc {
             doc_type,
             namespaces,
             mso,
-            prepared_sig,
+            mut prepared_sig,
         } = self;
 
-        let mut issuer_auth = prepared_sig.finalize(signature);
+        prepared_sig.0.signature = signature;
+        let mut issuer_auth = prepared_sig;
+        let val = Value::from_slice(&serde_cbor::to_vec(&x5chain.into_cbor()).unwrap()).unwrap();
```

Comment:
Update [this function](https://github.com/spruceid/isomdl/blob/main/src/issuance/x5chain.rs#L31) to return a `ciborium::Value` instead of a `serde_cbor::Value`, to avoid these uncessary conversions and unwraps.

Reply:
signature updated